docs/website/content/v0.3/en/configuration/v1alpha1.md

```diff
@@ -0,0 +1,276 @@
+```
+
+nolint: lll
+
+## Reference
+
+### type
+
+Type: `string`
+
+Valid Values:
+
+- ``init``
+- ``controlplane``
+- ``join``
+
+The machine's `type` defines the role of the machine within the cluster.
+
+#### Init
+
+Init node type designates the first control plane node to come up.
+You can think of it like a bootstrap node.
+This node will perform the initial steps to bootstrap the cluster -- generation of TLS assets, starting of the control plane, etc.
+
+#### Control Plane
+
+Control Plane node type designates the node as a control plane member.
+This means it will host etcd along with the Kubernetes master components such as API Server, Controller Manager, Scheduler.
+
+#### Worker
+
+Worker node type designates the node as a worker node.
+This means it will be an available compute node for scheduling workloads.
+
+### token
+
+Type: `string`
+
+The `token` is used by a machine to join the PKI of the cluster.
+Using this token, a machine will create a certificate signing request (CSR), and request a certificate that will be used as its' identity.
+
+Examples:
+
+```yaml
+token: 328hom.uqjzh6jnn2eie9oi
+```
+
+> Warning: It is important to ensure that this token is correct sinc a machine's certificate has a short TTL by default
+
+### ca
+
+Type: `PEMEncodedCertificateAndKey`
+
+The `ca` is the root certificate authority of the PKI.
+It is composed of a base64 encoded `crt` and `key`.
+
+Examples:
+
+```yaml
+ca:
+  crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJIekNCMHF...
+  key: LS0tLS1CRUdJTiBFRDI1NTE5IFBSSVZBVEUgS0VZLS0tLS0KTUM...
+
+```
+
+### certSANs
+
+Type: `array`
+
+The `certSANs` are extra certificate subject alternative names for the machine's certificate.
+By default, all non-loopback interface IPs are automatically added to the certificate's SANs.
+
+Examples:
+
+```yaml
+certSANs:
+  - 10.0.0.10
+  - 172.16.0.10
+  - 192.168.0.10
+
+```
+
+### kubelet
+
+Type: `KubeletConfig`
+
+The `kubelet` field is used to provide additional options to the kubelet.
+
+Examples:
+
+```yaml
+kubelet:
+  image:
+  extraArgs:
+    key: value
+
+```
+
+### network
+
+Type: `NetworkConfig`
+
+The `network` field is used to configure the machine's network.
+
+Examples:
+
+```yaml
+network:
+  hostname:
+  interfaces:
+  nameservers:
+
+```
+
+### install
+
+Type: `InstallConfig`
+
+The `install` field is used to provide instructions for bare-metal installations.
+
+Examples:
+
+```yaml
+install:
+  disk:
+  extraDiskArgs:
+  extraKernelArgs:
+  image:
+  bootloader:
+  wipe:
+  force:
+
+```
+
+### files
+
+Type: `array`
+
+The `files` field allows the addition of user specified files.
```

Comment:
probably want to clarify the contents should be a string/yaml block ( be explicit that contents shouldn't be b64 encoded ). i don't know why but in my head i assumed it was.. probably after seeing all the b64 encoded cert/key blobs